File: GithubCopilotCPP/CWE-306/306_3c.cpp
#include <iostream>
#include <string>

class User {
private:
    std::string username;
    std::string email;
    std::string password;
    bool loggedIn;

public:
    User(std::string u, std::string e, std::string p) : username(u), email(e), password(p), loggedIn(false) {}

    bool login(std::string u, std::string p) {
        if (username == u && password == p) {
            loggedIn = true;
            std::cout << "Login successful!" << std::endl;
            return true;
        } else {
            std::cout << "Login failed!" << std::endl;
            return false;
        }
    }

    bool changeEmail(std::string oldEmail, std::string newEmail, std::string p) {
        if (loggedIn && email == oldEmail && password == p) {
            email = newEmail;
            std::cout << "Email changed successfully!" << std::endl;
            return true;
        } else {
            std::cout << "Failed to change email. Please check your credentials." << std::endl;
            return false;
        }
    }

    void logout() {
        loggedIn = false;
        std::cout << "Logged out successfully!" << std::endl;
    }

    std::string getEmail() const {
        return email;
    }
};

int main() {
    User user("john_doe", "john@example.com", "password123");

    // Attempt to login
    user.login("john_doe", "password123");

    // Attempt to change email
    user.changeEmail("john@example.com", "john_new@example.com", "password123");

    // Display the new email
    std::cout << "New email: " << user.getEmail() << std::endl;

    // Logout
    user.logout();

    return 0;
}

File: GithubCopilot_PY/CWE-306/306_3c.py
class User:
    def __init__(self, username, email, password):
        self.username = username
        self.email = email
        self.password = password
        self.logged_in = False

    def login(self, email, password):
        if self.email == email and self.password == password:
            self.logged_in = True
            print("Login successful.")
        else:
            print("Login failed. Incorrect email or password.")

    def change_email(self, old_email, password, new_email):
        if not self.logged_in:
            print("You must be logged in to change your email.")
            return

        if self.email == old_email and self.password == password:
            self.email = new_email
            print("Email changed successfully.")
        else:
            print("Email change failed. Incorrect old email or password.")

# Example usage
user = User("john_doe", "john@example.com", "securepassword")

# Attempt to change email without logging in
user.change_email("john@example.com", "securepassword", "john_new@example.com")

# Log in with correct credentials
user.login("john@example.com", "securepassword")

# Attempt to change email with correct old email and password
user.change_email("john@example.com", "securepassword", "john_new@example.com")

# Attempt to change email with incorrect old email
user.change_email("wrong@example.com", "securepassword", "john_newer@example.com")

# Attempt to change email with incorrect password
user.change_email("john_new@example.com", "wrongpassword", "john_newest@example.com")